packages/tpetra/core/src/Tpetra_Util.hpp

```diff
@@ -514,12 +514,12 @@ namespace Tpetra {
 
     Kokkos::parallel_for("apply_permutation_1",
                         Kokkos::RangePolicy<Kokkos::DefaultHostExecutionSpace>(0, n),
-                        KOKKOS_LAMBDA(const int j) {
+                        [=](const int j) {
```

Comment:
@brian-kelley Is the problem here that `KOKKOS_LAMBDA` marks this as device accessible?

Reply:
Right, KOKKOS_LAMBDAs get compiled as both host and device code, but this calls ``std::vector`` methods which are only available in host code.